src/frontend/src/components/warnBox.ts

```diff
@@ -0,0 +1,36 @@
+import { html, TemplateResult } from "lit-html";
+import { warningIcon } from "./icons";
+
+interface warnBoxProps {
+  title: string;
+  message: string | TemplateResult;
+  componentType: string;
+  slot?: TemplateResult;
+}
+
+export const warnBox = (
+  props: warnBoxProps = {
+    title: "Warning",
+    message: html`Somewhere unsafe string that can also contain html
+      <b>Evil HTML</b>`,
+    componentType: "div",
+    slot: html`<button></button>`,
+  }
+): TemplateResult =>
+  html`${props.componentType === "div"
+    ? html` <div class="warnBox">
```

Comment:
the whole inner template is duplicated, which is not ideal. How about extracting that?